lib/ui/setup/pair_page.dart

```diff
@@ -1,148 +1,109 @@
-import 'dart:developer';
 import 'dart:ui';
 
 import 'package:cobble/domain/entities/pebble_scan_device.dart';
 import 'package:cobble/infrastructure/datasources/paired_storage.dart';
+import 'package:cobble/infrastructure/pigeons/pair_provider.dart';
 import 'package:cobble/infrastructure/pigeons/pigeons.dart';
+import 'package:cobble/infrastructure/pigeons/scan_provider.dart';
 import 'package:cobble/ui/common/icons/fonts/rebble_icons_stroke.dart';
 import 'package:cobble/ui/common/icons/watch_icon.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:shared_preferences/shared_preferences.dart';
 
-class PairPage extends StatefulWidget {
-  @override
-  State<StatefulWidget> createState() => new _PairPageState();
-}
-
 final ConnectionControl connectionControl = ConnectionControl();
 final ScanControl scanControl = ScanControl();
 
-class _PairPageState extends State<PairPage>
-    implements ScanCallbacks, PairCallbacks {
-  PairedStorage pairedStorage;
-
-  List<PebbleScanDevice> _pebbles = [];
-  bool _scanning = false;
-  bool _firstRun = false;
-
+class PairPage extends HookWidget {
   @override
-  void initState() {
-    super.initState();
-    ScanCallbacks.setup(this);
-    PairCallbacks.setup(this);
-    log("Prestart");
-    _isFirstRun();
-    scanControl.startBleScan();
-  }
+  Widget build(BuildContext context) {
+    final pairedStorage = useProvider(pairedStorageProvider);
+    final isFirstTime = useState(AsyncValue<bool>.loading());
+    final isFtu = isFirstTime.value.data;
```

Comment:
What is `Ftu`? I think `isFirstTime` and `isFtu`  names are a bit too similar and it's confusing which one is which.

Reply:
FTU = First time user.
This variable is just shorter version of isFirstTime, I can drop it and use isFirstTime.value.data in component instead of isFtu. 👍

Reply:
I renamed it to isFirstTimeState and isFirstTime.